### DISCUSSION TITLE
[A-L] (5 marzo 2024) Esercizio 4

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/progettazione-di-algoritmi#15] [Esercitazioni A-L]
(22.3-11,[1]). È possibile avere un vertice $u$ di un grafo diretto $G$ che finisce in un albero di ricerca in profondità che contiene solo $u$, anche se $u$ ha grado entrante e grado uscente almeno uno?

![image](https://github.com/sapienzastudentsnetwork/progalg2324/assets/40034429/46bdd3e0-105d-400b-a583-5a39aaf6ae62)

**Comment by Frisbon** [comment]
$x \rightarrow u  \rightarrow v$

Non può esistere un albero composto solo dal nodo $u$ perchè:
- anche se ignoriamo $x$ e cominciamo direttamente l'algoritmo da $u$, il DFS andrebbe automaticamente ad esplorare gli archi uscenti di $u$
- l'unico caso in cui $v$ non viene visitato è se VIS[ $v$ ] == 1, ma non è possibile perchè $\implies$ $v \in$ albero della DFS $\cancel\implies$ albero composto solo da $u$
